bin/check-azurerm-monitor-metric.rb

```diff
@@ -0,0 +1,305 @@
+
+                    res = Net::HTTP.start(uri.host, uri.port, use_ssl: false) do |http|
+                      req = Net::HTTP::Get.new(uri)
+                      req['Metadata'] = 'true'
+                      http.request(req)
+                    end
+
+                    handle_response(res)
+
+                    auth_resp = JSON.parse(res.body, symbolize_names: true)
+
+                    "#{auth_resp[:token_type]} #{auth_resp[:access_token]}"
+                  else
+                    provider = MsRestAzure::ApplicationTokenProvider.new(
+                      config[:tenant_id],
+                      config[:client_id],
+                      config[:client_secret]
+                    )
+
+                    provider.get_authentication_header
+                  end
+
+    begin
+      url = "https://management.azure.com#{resource}/providers/microsoft.insights/metrics?" \
+        "api-version=#{AZURE_API_VER}&" \
+        "metric=#{config[:metric]}&" \
+        "timespan=#{CGI.escape(timespan)}&" \
+        "aggregation=#{config[:aggregation]}"
+
+      url += "&$filter=#{CGI.escape(config[:filter])}" if config[:filter]
+
+      uri = URI.parse(url)
+
+      res = Net::HTTP.start(uri.host, uri.port, use_ssl: true) do |http|
+        req = Net::HTTP::Get.new(uri)
+        req['Authorization'] = auth_header
+        req['Content-Type'] = 'application/json'
+        http.request(req)
+      end
+    rescue Timeout::Error,
+           Errno::EINVAL, Errno::ECONNRESET,
+           EOFError, Net::HTTPBadResponse,
+           Net::HTTPHeaderSyntaxError, Net::ProtocolError,
+           Errno::ECONNREFUSED => e
+
+      critical e
+    end
+
+    handle_response(res)
+
+    JSON.parse(res.body, symbolize_names: true)
+  end
+
+  def resource
+    @resource ||= build_resource
+  end
+
+  def build_resource
+    if !config[:resource_name].to_s.empty?
+
+      if config[:resource_type].to_s.empty? ||
+         config[:resource_namespace].to_s.empty? ||
+         config[:resource_group].to_s.empty? ||
+         config[:subscription_id].to_s.empty?
+
+        unknown 'If resource type, namespace, or group is given, then all are required along with the subscription id.'
+      else
+        "/subscriptions/#{config[:subscription_id]}/resourceGroups/#{config[:resource_group]}/" \
+          "providers/#{resource_type}/#{config[:resource_name]}"
+      end
+    else
+      config[:resource_id].start_with?('/') ? config[:resource_id] : '/' + config[:resource_id]
+    end
+  end
+
+  def resource_type
+    if config[:resource_parent].to_s.empty?
+      "#{config[:resource_namespace]}/#{config[:resource_type]}"
+    else
+      "#{config[:resource_namespace]}/#{config[:resource_parent]}/#{config[:resource_type]}"
+    end
+  end
+
+  def timespan
+    # 10 min.  This should be enough time to capture the last value, without wasting API credits
+    start_date = Time.now.utc - 600
```

Comment:
I think that the timespan should be configurable, I can't say I know much about how azure does metrics but I do look at cloudwatch metrics in aws in shorter intervals than this for some metrics.